tools/pg_tools/pglite

```diff
@@ -0,0 +1,211 @@
+  case "$1" in
+    postgres)
+      ctl=pg_ctl
+      sql=psql
+      init=initdb
+      conf=postgresql.conf ;;
+    pipeline)
+      ctl=pipeline-ctl
+      sql=pipeline
+      init=pipeline-init
+      conf=pipelinedb.conf ;;
+    *) msg "Unrecognized personality: $1"
+       err 'Please choose from: pipeline postgres'
+  esac
+}
+
+function load_personality {
+  if [[ -f "$root"/personality ]]
+  then set_personality "$(< "$root"/personality)"
+  fi
+}
+
+function setup {
+  if [[ -d "$root" ]]
+  then msg "NB: Skipping setup; dir exists: $root"
+  else initialize
+  fi
+}
+
+function connect {
+  load_personality
+  if ! is_running
+  then
+    ctl start -w
+    trap finalize EXIT
+  fi
+  sql "$@"
+}
+
+function url {
+  out "postgres://$role@[$(absroot)]/$role"
+}
+
+function absroot {
+  cd "$root" && pwd -P
+}
+
+function init_cluster_directory {
+  local tmp=/tmp/pglite~"$(whoami)".$$ abs=
+
+  "$init" -D "$root"/db -A trust --locale en_US.UTF-8
+  abs="$(absroot)"
+
+  ( trap 'rm -rf "$tmp"' EXIT
+    mkdir -p "$tmp"
+    sed "s|^#*unix_socket_directory = .*$|unix_socket_directory = '"$abs"'|
+         s|^#*unix_socket_directories = .*$|unix_socket_directories = '"$abs"'|
+         s|^#*listen_addresses = .*$|listen_addresses = ''|
+         s|^#*datestyle = .*$|datestyle = 'iso, ymd'|
+         s|^#*intervalstyle = .*$|intervalstyle = 'iso_8601'|
+         s|^#*timezone = .*$|timezone = 'UTC'|
+        " "$root"/db/"$conf" > "$tmp"/conf
+
+    cat "$tmp"/conf > "$root"/db/"$conf" )
+
+  out "$personality" > "$root"/personality
+}
+
+function init_database_and_user {
+sql_admin -v ON_ERROR_STOP=on <<SQL
+  CREATE ROLE "$role" WITH SUPERUSER LOGIN REPLICATION;
+  CREATE DATABASE "$role" WITH OWNER "$role";
+SQL
+}
+
+function ctl {
+  load_personality
+  "$ctl" -D "$root"/db -l "$root"/log "$@"
+}
+
+function sql {
+  "$sql" -h "$(absroot)" --dbname="user=$role dbname=$role" "$@"
+}
+
+function sql_admin {
+  "$sql" -h "$(absroot)" --dbname="user=$admin dbname=template1" "$@"
+}
+
+function finalize {
+  local code=$?
+  ctl stop -m fast 2>/dev/null || true
+  kill %1 2>/dev/null || true
+  exit $code
+}
+
+function initialize {
+  init_cluster_directory
+  trap finalize EXIT
+  touch "$root"/log
+  tail -qF "$root"/log &
+  ctl start -w
+  init_database_and_user
+  ctl stop -w
+  kill %1
+  wait %1 2>/dev/null || true
+}
+
+function is_running {
+  ctl status | fgrep -q 'server is running'
+}
+
+function clean {
+  msg "Removing: $root"
+  [[ ! -e "$root" ]] || rm -r "$root"
+}
+
+##################################################################### Utilities
+
+function msg { out "$*" >&2 ;}
+function err { local x=$? ; msg "$*" ; return $(( $x == 0 ? 1 : $x )) ;}
+function out { printf '%s\n' "$*" ;}
+
+# Handles "no-match" exit code specified by POSIX for filtering tools.
+function maybe { "$@" || return $(( $? == 1 ? 0 : $? )) ;}
+
+
+######################### Delegates to subcommands or runs main, as appropriate
+
+main "$@"
```

Comment:
We need to document the license / source+license for this script somewhere, i.e. https://github.com/solidsnack/pglite/blob/master/LICENSE

Reply:
I don't think we should ship this tool at all, we should rather depend on it.

Reply:
works for me!

Reply:
removed this file and we now depend on a conda package